test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/crypto/HapiCryptoTransfer.java

```diff
@@ -393,12 +396,19 @@ protected HapiCryptoTransfer self() {
 	}
 
 	private List<TokenTransferList> transfersForNft(HapiApiSpec spec) {
+		var uniqueCount = tokenAwareProviders.stream()
+				.filter(Predicate.not(TokenMovement::isFungibleToken)).count();
 		Map<TokenID, List<NftTransfer>> aggregated = tokenAwareProviders.stream()
 				.filter(Predicate.not(TokenMovement::isFungibleToken))
 				.map(p -> p.specializedForNft(spec))
-				.collect(groupingBy(
+				.collect(Collectors.toMap(
 						TokenTransferList::getToken,
-						flatMapping(xfers -> xfers.getNftTransfersList().stream(), toList())));
+						TokenTransferList::getNftTransfersList,
+						(left, right) -> Stream.of(left, right).flatMap(Collection::stream).collect(toList()),
+						LinkedHashMap::new));
+		if (aggregated.size() != 0 && uniqueCount != aggregated.size()) {
+			throw new RuntimeException("Duplicated token key!");
```

Comment:
I think that it would be better if we add a more descriptive message. If you see "Duplicated token key!" in the e2e you wouldn't know what exactly went wrong. Maybe a better message could be "Duplicated Token Id set in unique token movement"